src/multiscale/multiscale.jl

```diff
@@ -1,33 +1,131 @@
-function multiscale(alg::MultiScaleAlgorithm, e::ComplexityEstimator, x::AbstractVector;
-        maxscale::Int = 8)
+# This file contains an API for multiscale (coarse-grained/downsampled) computations.
+# This API is not final nor agreed upon yet. Nothing is exported or documented
+# as part of public API. It must be considered as work in progress.
+export downsample
+export multiscale, multiscale_normalized
+export MultiScaleAlgorithm
 
-    downscaled_timeseries = [downsample(alg, s, x) for s in 1:maxscale]
-    return complexity.(Ref(e), downscaled_timeseries)
-end
+using Statistics
+
+"""
+    MultiScaleAlgorithm
+
+The supertype for all multiscale coarse-graining/downsampling algorithms. Concrete subtypes
+are:
+
+- [`RegularDownsampling`](@ref)
+- [`CompositeDownsampling`](@ref)
+"""
+abstract type MultiScaleAlgorithm end
+
+"""
+    downsample(algorithm::MultiScaleAlgorithm, s::Int, x)
+
+Downsample and coarse-grain `x` to scale `s` according to the given
+[`MultiScaleAlgorithm`](@ref). The return type depends on `algorithm`.
+"""
+downsample(method::MultiScaleAlgorithm, s::Int, x)
+
+downsample(alg::MultiScaleAlgorithm,  s::Int, x::AbstractStateSpaceSet) =
+    StateSpaceSet(map(t -> downsample(alg, s, t)), columns(x)...)
+
+"""
+    multiscale(algorithm::MultiScaleAlgorithm, [args...], x; maxscale::Int = 8)
```

Comment:
Hm, I think this is a bit in harmonious, to have `maxscale` as a keyword instead of an input argument of the multiscale algorithm.  I think it overall keeps a clean interface if `maxscale` is given to the algorithm. In all other "core" functions of the package, there are no keyword arugments, everything is given as arguments to the "configuration" types.